### PR TITLE
usm: tls: tests without a tracer

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

The PR allows running TLS monitoring tests using USM monitor, without the old requirement of having a full Tracer.
The PR simplifies tests and moves them back to USM packages.

### Motivation
Simplify tests

The PR https://github.com/DataDog/datadog-agent/pull/21259 was a preliminary step, and now USM owns all kernel related code that once was in NPM

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/protocols/tls/gotls/testutil/helpers.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/DataDog/datadog-agent/pkg/network/config"
+	"github.com/DataDog/datadog-agent/pkg/network/protocols/http"
+)
+
+// isFedora returns true if the current OS is Fedora.
+// go-tls does not work correctly on Fedora 35, 36, 37 and 38.
+func isFedora() bool {
+	info, err := host.Info()
+	if err != nil {
+		return false
```
Comment (amitslavin) [inline pkg/network/protocols/tls/gotls/testutil/helpers.go:22]: You will return false instead of the error, and you would not know if you were able to get the info, or it is version that is different from the options you checked.

File: pkg/network/protocols/tls/gotls/testutil/helpers.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	}
+	return info.Platform == "fedora" && (info.PlatformVersion == "35" || info.PlatformVersion == "36" || info.PlatformVersion == "37" || info.PlatformVersion == "38")
```
Comment (amitslavin) [inline pkg/network/protocols/tls/gotls/testutil/helpers.go:24]: maybe create const with the supported versions, and rename it to var with supported versions. that would make it nicer and more readable

File: pkg/network/tracer/usm_http2_monitor_test.go [hunk 3]
```diff
@@ -203,7 +188,7 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
 			},
 		},
 		{
-			name: "validate literal header field never indexed",
+			name: "guy validate literal header field never indexed",
```
Comment (amitslavin) [inline pkg/network/tracer/usm_http2_monitor_test.go:186]: guy again?

File: pkg/network/tracer/usm_http2_monitor_test.go [hunk 4]
```diff
@@ -219,7 +204,7 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
 			},
 		},
 		{
-			name: "validate path with index 4",
+			name: "guy validate path with index 4",
```
Comment (amitslavin) [inline pkg/network/tracer/usm_http2_monitor_test.go:202]: guy again?

File: pkg/network/usm/monitor_test.go [hunk 5]
```diff
@@ -922,30 +931,38 @@ func getClientsArray(t *testing.T, size int) []*nethttp.Client {
 	return res
 }
 
-func startH2CServer(t *testing.T) {
-	t.Helper()
-
+func startH2CServer(address string, isTLS bool) (func(), error) {
 	srv := &nethttp.Server{
-		Addr: http2SrvPortStr,
+		Addr: authority,
```
Comment (amitslavin) [inline pkg/network/usm/monitor_test.go:936]: why did you change it to authority?

File: pkg/network/tracer/tracer_usm_linux_test.go [hunk 6]
```diff
@@ -82,6 +82,27 @@ func classificationSupported(config *config.Config) bool {
...
+func setupUSMTLSMonitor(t *testing.T, cfg *config.Config) *usm.Monitor {
+	usmMonitor, err := usm.NewMonitor(cfg, nil, nil, nil)
+	require.NoError(t, err)
+	require.NoError(t, usmMonitor.Start())
+	t.Cleanup(usmMonitor.Stop)
+	t.Cleanup(utils.ResetDebugger)
+	return usmMonitor
+}
+
+func getHTTPLikeProtocolStats(monitor *usm.Monitor, protocolType protocols.ProtocolType) map[http.Key]*http.RequestStats {
```
Comment (amitslavin) [inline pkg/network/tracer/tracer_usm_linux_test.go]: please add doc

Comment (amitslavin) [reply pkg/network/tracer/tracer_usm_linux_test.go]: I do think that the name of the function is really not clear.

**Review by amitslavin — CHANGES_REQUESTED** [review]

**Comment by amitslavin** [comment]
General comment: I do not like the fact that after the change TestRawTraffic is different from all other tests in its way we run it. Do you want to change it as part of this PR? or I would draft one?

File: pkg/network/protocols/tls/gotls/testutil/helpers.go (hunk 1 above)
Comment (guyarb) [reply pkg/network/protocols/tls/gotls/testutil/helpers.go:22]: changed to match https://github.com/DataDog/datadog-agent/pull/22199/files#diff-1b200ed6e29617e4a690bf1fbaf8694c2420b81bfa968c1a7ae221f36323c293R769-R775

File: pkg/network/protocols/tls/gotls/testutil/helpers.go (hunk 2 above)
Comment (guyarb) [reply pkg/network/protocols/tls/gotls/testutil/helpers.go:24]: changed

File: pkg/network/tracer/usm_http2_monitor_test.go (hunk 3 above)
Comment (guyarb) [reply pkg/network/tracer/usm_http2_monitor_test.go:186]: removed

File: pkg/network/tracer/usm_http2_monitor_test.go (hunk 4 above)
Comment (guyarb) [reply pkg/network/tracer/usm_http2_monitor_test.go:202]: removed

File: pkg/network/usm/monitor_test.go (hunk 5 above)
Comment (guyarb) [reply pkg/network/usm/monitor_test.go:936]: to ensure we're tide to the `lo` interface, and not to all interfaces

File: pkg/network/tracer/tracer_usm_linux_test.go (hunk 6 above)
Comment (guyarb) [reply pkg/network/tracer/tracer_usm_linux_test.go]: added documentation

**Review by amitslavin — APPROVED** [review]
